qualang_tools/wirer/wirer/channel_specs.py

```diff
@@ -0,0 +1,150 @@
+                 in_port: int = None, out_port: int = None):
+
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelMwFemInput(con=con, slot=slot, port=in_port),
+            InstrumentChannelMwFemOutput(con=con, slot=slot, port=out_port)
+        ]
+
+class ChannelSpecLfFemSingle(ChannelSpec):
+    def __init__(self, con: int = None,
+                 in_slot: int = None,in_port: int = None,
+                 out_slot: int = None, out_port: int = None):
+
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelLfFemInput(con=con, slot=in_slot, port=in_port),
+            InstrumentChannelLfFemOutput(con=con, slot=out_slot, port=out_port)
+        ]
+
+
+class ChannelSpecLfFemBaseband(ChannelSpec):
+    def __init__(self, con: int = None, slot: int = None,
+                 in_port_i: int = None, in_port_q: int = None,
+                 out_port_i: int = None, out_port_q: int = None):
+
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelLfFemInput(con=con, slot=slot, port=in_port_i),
+            InstrumentChannelLfFemInput(con=con, slot=slot, port=in_port_q),
+            InstrumentChannelLfFemOutput(con=con, slot=slot, port=out_port_i),
+            InstrumentChannelLfFemOutput(con=con, slot=slot, port=out_port_q)
+        ]
+
+
+class ChannelSpecOpxPlusSingle(ChannelSpec):
+    def __init__(self, con: int = None, in_port: int = None, out_port: int = None):
+
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelOpxPlusInput(con=con,  port=in_port),
+            InstrumentChannelOpxPlusOutput(con=con, port=out_port)
+        ]
+
+
+class ChannelSpecOpxPlusBaseband(ChannelSpec):
+    def __init__(self, con: int = None,
+                 in_port_i: int = None, in_port_q: int = None,
+                 out_port_i: int = None, out_port_q: int = None):
+
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelOpxPlusInput(con=con, port=in_port_i),
+            InstrumentChannelOpxPlusInput(con=con, port=in_port_q),
+            InstrumentChannelOpxPlusOutput(con=con, port=out_port_i),
+            InstrumentChannelOpxPlusOutput(con=con, port=out_port_q)
+        ]
+
+
+class ChannelSpecOctave(ChannelSpec):
+    def __init__(self, index: int = None, rf_out: int = None, rf_in: int = None):
+        super().__init__()
+        self.channel_templates = [
+            InstrumentChannelOctaveInput(con=index, port=rf_in),
+            InstrumentChannelOctaveOutput(con=index, port=rf_out),
+        ]
+
+
+class ChannelSpecLfFemBasebandAndOctave(ChannelSpec):
+    def __init__(self,
+                 con: int = None, slot: int = None,
+                 in_port_i: int = None, in_port_q: int = None,
+                 out_port_i: int = None, out_port_q: int = None,
+                 octave_index: int = None, rf_out: int = None, rf_in: int = None):
```

Comment:
[imo]

A very minor point, it would be less confusing to flip `rf_out` and `rf_in` as 
```
octave_index: int = None, rf_in: int = None, rf_out: int = None, 
```
in case somebody give arguments without arg names explicitly.

Reply:
Great points Hiro, thank you. Will do this

Reply:
Done :)